### ISSUE TITLE
BERT Classifier: Loss is 0?

### ISSUE THREAD

**Issue by bigabig** [asyml/texar-pytorch#62]
Hey,

I am trying to run the BERT Classifier example with the MRPC task. 

I get the following error:
```
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm 2018.3.3\helpers\pydev\pydevd.py", line 1741, in <module>
    main()
  File "C:\Program Files\JetBrains\PyCharm 2018.3.3\helpers\pydev\pydevd.py", line 1735, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "C:\Program Files\JetBrains\PyCharm 2018.3.3\helpers\pydev\pydevd.py", line 1135, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:\Program Files\JetBrains\PyCharm 2018.3.3\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "C:/Development/Git/texar-pytorch/examples/bert/bert_classifier_main.py", line 274, in <module>
    main()
  File "C:/Development/Git/texar-pytorch/examples/bert/bert_classifier_main.py", line 258, in main
    _train_epoch()
  File "C:/Development/Git/texar-pytorch/examples/bert/bert_classifier_main.py", line 156, in _train_epoch
    input_ids = batch["input_ids"]
  File "C:\Development\Git\texar-pytorch\texar\data\data\dataset_utils.py", line 78, in __getitem__
    return self._batch[item]
KeyError: 'input_ids'
```

Changing the keys as for example in input_ids = batch["input_ids"] to input_ids = batch["data_input_ids"] seems to fix these errors.

However, then i tried the script like this 
`python bert_classifier_main.py --do_train --do_eval`

and get the following logs:
```
python bert_classifier_main.py --do_train --do_eval
Pretrained model loaded from bert_pretrained_models/uncased_L-12_H-768_A-12\bert_model.ckpt
INFO:root:step: 50; loss: 0
INFO:root:step: 100; loss: 0
INFO:root:step: 150; loss: 0
INFO:root:step: 200; loss: 0
INFO:root:step: 250; loss: 0
INFO:root:step: 300; loss: 0
INFO:root:eval accu: 0.8578; loss: 0.3904; nsamples: 408
```
Is this correct?

**Comment by huzecong** [comment]
The `KeyError` was the result of the addition of a new data module; we'll fix this soon. Just to confirm, have you also added the `"data_"` prefix to `"segment_ids"` and `"label_ids"`?

For the loss part, I'm not sure what happened. This should not be correct. We're still looking into this problem.

**Comment by bigabig** [comment]
Hey, yes I added the "data_" prefix to all related dataset keys (segment_ids, label_ids, input_ids). After that I could train the model with the script. Not sure what happen with the loss, though.

**Comment by haoransh** [comment]
Is the loss still 0 after you change these keys? I'm not very sure which commit induces this bug but hopefully, [this version](https://github.com/asyml/texar-pytorch/commit/ac3e334e491f524dd01654b07af030fa20c88b34) can work. We're still working on this.

**Comment by bigabig** [comment]
hey, i changed the keys and I did
```
git checkout ac3e334e491f524dd01654b07af030fa20c88b34
pip install -e .
```
to get to the commit you mentioned.

This is the output of the training. So, Loss is still 0. 
```
(env) C:\Development\Git\texar-pytorch\examples\bert>python bert_classifier_main.py --do_train --do_eval
Pretrained model loaded from bert_pretrained_models/uncased_L-12_H-768_A-12\bert_model.ckpt
INFO:root:step: 50; loss: 0
INFO:root:step: 100; loss: 0
INFO:root:step: 150; loss: 0
INFO:root:step: 200; loss: 0
INFO:root:step: 250; loss: 0
INFO:root:step: 300; loss: 0
INFO:root:eval accu: 0.8480; loss: 0.4097; nsamples: 408
```

**Comment by haoransh** [comment]
Sorry you may want to try this version:
https://github.com/asyml/texar-pytorch/commit/15d04a7ad6c7432ad2e0b9accf434f53be423459#diff-7ab4b40bab7f699ffd21cacea48a51aa

It should be able to give such results:
```
INFO:root:step: 50; loss: 0.5814850926399231;
```

**Comment by huzecong** [comment]
This issue is now fixed in 17c44ac. There are actually no bugs in the model -- it's the logging message that was wrong. A `%d` specifier was incorrectly used for the floating point variable loss (which was less than 1).

Sorry for the inconvenience.
